types/specification.d.ts

```diff
@@ -0,0 +1,151 @@
+      staticProperties?: PropertyMap & ThisType<F>;
+      /** A set of static properties that will be added to the `Stamp` by deep property merge. */
+      staticDeepProperties?: PropertyMap & ThisType<F>;
+      /** A set of object property descriptors (`PropertyDescriptor`) to apply to the `Stamp`. */
+      staticPropertyDescriptors?: PropertyDescriptorMap & ThisType<F>;
+      /** An array of functions that will run in sequence while creating an object instance from a `Stamp`. `Stamp` details and arguments get passed to initializers. */
+      initializers?: Array<Initializer<I>>;
+      /** An array of functions that will run in sequence while creating a new `Stamp` from a list of `Composable`s. The resulting `Stamp` and the `Composable`s get passed to composers. */
+      composers?: Array<Composer<I>>;
+      /** A set of options made available to the `Stamp` and its initializers during object instance creation. These will be copied by assignment. */
+      configuration?: PropertyMap & ThisType<F>;
+      /** A set of options made available to the `Stamp` and its initializers during object instance creation. These will be deep merged. */
+      deepConfiguration?: PropertyMap & ThisType<F>;
+  }
+
+  /**
+   * A function used as `.initializers` argument.
+   * @param {type} I The object type produced by the `Stamp`.
+   * @param {PropertyMap} options The `options` argument passed into the stamp, containing properties that may be used by initializers.
+   * (Cf. the `Descriptor` attributes `configuration` and `deepConfiguration`)
+   * @param {InitializerContext} context (optional) The context of current `.initializers` function.
+   * @returns {I} The object produced by the `Stamp`.
+   */
+  interface Initializer<I> {
+      // TODO: check that the value of `this` is correct
+      (/*this: I, */options?: PropertyMap, context?: InitializerContext<I>): I;
+  }
+
+  /**
+   * The `Initializer` function context.
+   * @param {type} I The object type produced by the `Stamp`.
+   * @param {I} instance (optional) The object instance being produced by the `Stamp`. If the initializer returns a value other than `undefined`, it replaces the instance.
+   * @param {Stamp<I>} stamp (optional) A reference to the `Stamp` producing the instance.
+   * @param {Array<any>} args (optional) An array of the arguments passed into the `Stamp`, including the options argument.
+   */
+  interface InitializerContext<I> {
+      /** The object instance being produced by the `Stamp`. If the initializer returns a value other than `undefined`, it replaces the instance. */
+      instance?: I;
+      /** A reference to the `Stamp` producing the instance. */
+      // TODO: investigate how to support extended Stamp interfaces
+      stamp?: Stamp<I>;
+      /** An array of the arguments passed into the `Stamp`, including the options argument. */
+      // ! above description from the specification is obscure
+      args?: Array<any>;
```

Comment:
I am unsure if `args` is properly typed. The specification is not very clear here.